fv/policysync_test.go

```diff
@@ -0,0 +1,511 @@
+			// permissions on the socket so the test process can connect.
+			felix.Exec("chmod", "a+rw", "/var/run/calico/policy-mgmt.sock")
+			mgmtClient = nodeagentmgmt.ClientUds(hostMgmtSocketPath)
+		})
+
+		var (
+			hostWlSocketPath, containerWlSocketPath [3]string
+		)
+
+		dirNameForWorkload := func(wl *workload.Workload) string {
+			return "ps-" + wl.WorkloadEndpoint.Spec.Pod
+		}
+
+		createWorkloadDirectory := func(wl *workload.Workload) (string, string) {
+			dirName := dirNameForWorkload(wl)
+			hostWlDir := tempDir + "/" + dirName
+			os.MkdirAll(hostWlDir, 0777)
+			return hostWlDir, "/var/run/calico/" + dirName
+		}
+
+		sendCreate := func(wl *workload.Workload) (*mgmtintf_v1.Response, error) {
+			dirName := dirNameForWorkload(wl)
+
+			resp, err := mgmtClient.WorkloadAdded(&mgmtintf_v1.WorkloadInfo{
+				Attrs: &mgmtintf_v1.WorkloadInfo_WorkloadAttributes{
+					Uid:       wl.WorkloadEndpoint.Spec.Pod,
+					Namespace: "fv",
+					Workload:  wl.WorkloadEndpoint.Spec.Pod,
+				},
+				Workloadpath: dirName,
+			})
+			log.WithField("response", resp).Info("WorkloadAdded response")
+			return resp, err
+		}
+
+		Context("after creating a client for each workload", func() {
+			BeforeEach(func() {
+				for i, wl := range w {
+					// Create the workload directory, this would normally be the responsibility of the
+					// flex volume driver.
+					hostWlDir, containerWlDir := createWorkloadDirectory(wl)
+					hostWlSocketPath[i] = hostWlDir + "/policysync.sock"
+					containerWlSocketPath[i] = containerWlDir + "/policysync.sock"
+
+					// Tell Felix about the new directory.
+					_, err := sendCreate(wl)
+					Expect(err).NotTo(HaveOccurred())
+				}
+			})
+
+			It("felix should create the workload socket", func() {
+				for _, p := range hostWlSocketPath {
+					Eventually(p).Should(BeAnExistingFile())
+				}
+			})
+
+			Context("with open workload connections", func() {
+
+				// Then connect to it.
+				var (
+					wlConn   [3]*grpc.ClientConn
+					wlClient [3]proto.PolicySyncClient
+					cancel   context.CancelFunc
+					ctx      context.Context
+					err      error
+				)
+
+				createWorkloadConn := func(i int) (*grpc.ClientConn, proto.PolicySyncClient) {
+					var opts []grpc.DialOption
+					opts = append(opts, grpc.WithInsecure())
+					opts = append(opts, grpc.WithDialer(unixDialer))
+					var conn *grpc.ClientConn
+					conn, err = grpc.Dial(hostWlSocketPath[i], opts...)
+					Expect(err).NotTo(HaveOccurred())
+					wlClient := proto.NewPolicySyncClient(conn)
+					return conn, wlClient
+				}
+
+				BeforeEach(func() {
+					ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+
+					for i := range w {
+						// Use the fact that anything we exec inside the Felix container runs as root to fix the
+						// permissions on the socket so the test process can connect.
+						Eventually(hostWlSocketPath[i]).Should(BeAnExistingFile())
+						felix.Exec("chmod", "a+rw", containerWlSocketPath[i])
```

Comment:
Agreed, I think ti's covered by one of the JIRA tickets (to finalise the config options, IIRC).  I pulled forward some of that work into this sprint so it would hang together for this testing but the config isn't "done done" so I didn't pull in the story itself.